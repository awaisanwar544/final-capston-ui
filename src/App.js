import React, { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';

import {
  BrowserRouter as Router,
} from 'react-router-dom';

import SplashPage from './pages/SplashPage';

import AppRoutes from './components/AppRoutes';

import { getProviders } from './redux/reducers/providers';

function App() {
  const dispatch = useDispatch();
  const localUser = localStorage.getItem('user');
  const user = JSON.parse(localUser);
  const renderSplash = (window.location.pathname === '/resetpassword') ? false : !user;
  const [splash, setSplash] = useState(renderSplash);

  const removeSplash = () => {
    setSplash(false);
  };

  useEffect(() => {
    dispatch(getProviders());
  }, []);

  return (
    <>
      {splash
        ? (
          <>
            <Router>
              <SplashPage removeSplash={removeSplash} />
            </Router>
          </>
        )
        : (
          <>
            <Router>
              <AppRoutes />
            </Router>
          </>
        )}
    </>
  );
}

export default App;
