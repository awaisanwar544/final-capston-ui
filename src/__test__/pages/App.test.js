import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import App from '../../App';
import store from '../../redux/store';

describe('App', () => {
  test('Render App', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
