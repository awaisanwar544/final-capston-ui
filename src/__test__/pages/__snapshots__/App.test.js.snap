// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Render App 1`] = `
<div
  className="flex flex-col mt-20 items-center"
  style={
    Object {
      "opacity": 0,
      "transform": "scale(10) translateZ(0)",
    }
  }
>
  <div
    className="flex w-full p-10 justify-center"
  >
    <a
      href="/"
    >
      <img
        alt="logo"
        className="w-48"
        src="logo.png"
      />
    </a>
  </div>
  <div
    className="flex items-center justify-center space-x-3"
  >
    <a
      className="bg-custom-white-500 text-custom-grey-500 border border-custom-green-500 hover:bg-custom-green-500 hover:text-white font-bold py-2 px-4 rounded focus:outline-none"
      exact="true"
      href="/signin"
      onClick={[Function]}
    >
      Sign In
    </a>
    <a
      className="bg-custom-white-500 text-custom-grey-500 border border-custom-green-500 hover:bg-custom-green-500 hover:text-white font-bold py-2 px-4 rounded focus:outline-none"
      exact="true"
      href="/signup"
      onClick={[Function]}
    >
      Register
    </a>
  </div>
</div>
`;
